Support SymInt placeholder in wrapper fxir

nandesuka opened the pull request:
Summary:
add support for symint placeholders

added two test cases with dynamic reshape
- dynamic info coming from tmd on placeholders
- dynamic info coming from placeholders (symints)

Test Plan:
test_reshape_dynamic_ph
test_reshape_dynamic_tmd

Differential Revision: D86984100




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

blaine-rister reviewed (approved):
LGTM, but I left a few style comments. I think there are a few linter issues as well.

angelayi commented on torch/_inductor/compile_fx.py:
```diff
@@ -2537,23 +2537,26 @@ def _extract_inputs_from_exported_gm(
-                        f"{fi.device} vs {i.device}. If the model was exported via torch.export(), "
-                        "make sure torch.export() and torch.aot_compile() run on the same device."
-                    )
+                if isinstance(fi, torch.Tensor) and isinstance(i, torch.Tensor):
```
nit: 
```python
if fi is not None and isinstance(fi, torch.Tensor):
    assert isinstance(i, torch.Tensor)
```

nandesuka committed:
Support SymInt placeholder in wrapper fxir (#167757)

Summary:

add support for symint placeholders

added two test cases with dynamic reshape
- dynamic info coming from tmd on placeholders
- dynamic info coming from placeholders (symints)

Test Plan:
test_reshape_dynamic_ph
test_reshape_dynamic_tmd

Reviewed By: blaine-rister

Differential Revision: D86984100
